.gitignore

```diff
@@ -5,3 +5,4 @@ vendor
 classes/log/error_log.txt
 composer.phar
 .idea/
+lib/*/docs/examples/*_test.json
```

Comment:
Ah. You've highlighted a small issue, actually, with the generation of those JSON files. They're not actually supposed to have the `_test` extension:

```php
    protected function create_example_file($output) {
        $classarray = explode('\\', get_class($this));
        $eventname = str_replace('Test', '', array_pop($classarray));
        $examplefile = __DIR__ . '/../../../lib/emitter/docs/examples/' . $eventname . '.json';
        file_put_contents($examplefile, json_encode($output, JSON_PRETTY_PRINT));
    }
```

I should have changed the `str_replace` line to take into consideration the change of filename.

I'll sort that out now.